packages/framework/src/servers/nest/nest.module.ts

```diff
@@ -43,7 +44,7 @@ export class NovuModule extends NovuBaseModule {
   static register(options: typeof OPTIONS_TYPE, customProviders?: Provider[]) {
     const superModule = super.register(options);
 
-    superModule.controllers = [NovuController];
+    superModule.controllers = [applyDecorators(NovuController, options.controllerDecorators || [])];
     superModule.providers?.push(registerApiPath, NovuClient, NovuHandler, ...(customProviders || []));
     superModule.exports = [NovuClient, NovuHandler];
 
```

Comment:
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
⚠ **issue:** I can't add a comment on it because there's no diff, but we also need to support the injected `controllerDecorators` on the `registerAsync` method too to ensure consistency on both APIs.‏

Reply:
Good point! I couldn't follow the options logic or where they are defined. Should it come just from the useFactory?

Reply:
I had some trouble here too when I implemented the NestJS Framework module. I think it will require something like calling the `super.registerAsync` method and accessing the defined options on the super class, but I'm not certain.

Reply:
I will check this on my local soon.

Reply:
Check out the NestJS [`HttpModule`](https://github.com/nestjs/axios/blob/457584e0a1673f6fae44d7be8e246a50525f7f6f/lib/http.module.ts#L60), they've implemented their `registerAsync` a little more flexibly without using the NestJS built-in `ConfigurableModuleBuilder`, which grants access to async options more easily.

This is the approach we need to take for the `customProviders` and `controllerDecorators` properties. We can then deprecate the second method parameter which supplies only `customProviders` in favour of a single configuration object for both `register` + `registerAsync`

Reply:
Couldn't make it work :| I'm going to merge this PR as I need to test the Stripe behaviour on staging, and will raise a separate PR for the registerAsync behaviour later today.

Reply:
Ok sounds good, I'll publish Framework from 1 commit this branch's merge commit to `next` to avoid releasing this change in that case. We should publish the feature in totality alongside the docs.